src/nnet3/nnet-am-decodable-simple.h

```diff
@@ -0,0 +1,174 @@
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef KALDI_NNET3_NNET_AM_DECODABLE_SIMPLE_H_
+#define KALDI_NNET3_NNET_AM_DECODABLE_SIMPLE_H_
+
+#include <vector>
+#include "base/kaldi-common.h"
+#include "gmm/am-diag-gmm.h"
+#include "hmm/transition-model.h"
+#include "itf/decodable-itf.h"
+#include "nnet3/nnet-optimize.h"
+#include "nnet3/nnet-compute.h"
+#include "nnet3/am-nnet-simple.h"
+
+namespace kaldi {
+namespace nnet3 {
+
+
+struct DecodableAmNnetSimpleOptions {
+  int32 frames_per_chunk;
+  BaseFloat acoustic_scale;
+  bool debug_computation;
+  NnetOptimizeOptions optimize_config;
+  NnetComputeOptions compute_config;
+
+  DecodableAmNnetSimpleOptions():
+      frames_per_chunk(50),
+      acoustic_scale(0.1) { }
+
+  void Register(OptionsItf *opts) {
+    opts->Register("frames-per-chunk", &frames_per_chunk,
+                   "Number of frames in each chunk that is separately evaluated "
+                   "by the neural net.");
+    opts->Register("acoustic-scale", &acoustic_scale,
+                   "Scaling factor for acoustic log-likelihoods");
+    opts->Register("debug-computation", &debug_computation, "If true, turn on "
+                   "debug for the actual computation (very verbose!)");
+
+    // register the optimization options with the prefix "optimization".
+    ParseOptions optimization_opts("optimization", opts);
+    optimize_config.Register(&optimization_opts);
+
+    // register the compute options with the prefix "computation".
+    ParseOptions compute_opts("computation", opts);
+    compute_config.Register(&compute_opts);
+  }
+};
+
+/* DecodableAmNnetSimple is a decodable object that decodes with a neural net
+   acoustic model of type AmNnetSimple.  It can accept just input features, or
+   input features plus iVectors.
+*/
+class DecodableAmNnetSimple: public DecodableInterface {
+ public:
+  /// Constructor that just takes the features as input, but can also optionally
+  /// take batch-mode or online iVectors.  Note: it stores references to all
+  /// arguments to the constructor, so don't delete them till this goes out of
+  /// scope.
+
+  DecodableAmNnetSimple(const DecodableAmNnetSimpleOptions &opts,
+                        const TransitionModel &trans_model,
+                        const AmNnetSimple &am_nnet,
+                        const MatrixBase<BaseFloat> &feats,
+                        const VectorBase<BaseFloat> *ivector = NULL,
+                        const MatrixBase<BaseFloat> *online_ivectors = NULL,
+                        int32 online_ivector_period = NULL);
```

Comment:
Yes it is-- I'll fix it next time I commit.  Thanks.

On Thu, Aug 20, 2015 at 6:35 PM, Xingyu Na <notifications@github.com> wrote:

> In src/nnet3/nnet-am-decodable-simple.h
> <https://github.com/kaldi-asr/kaldi/pull/83#discussion_r37599543>:
>
> > +   input features plus iVectors.
> > +*/
> > +class DecodableAmNnetSimple: public DecodableInterface {
> > + public:
> > +  /// Constructor that just takes the features as input, but can also optionally
> > +  /// take batch-mode or online iVectors.  Note: it stores references to all
> > +  /// arguments to the constructor, so don't delete them till this goes out of
> > +  /// scope.
> > +
> > +  DecodableAmNnetSimple(const DecodableAmNnetSimpleOptions &opts,
> > +                        const TransitionModel &trans_model,
> > +                        const AmNnetSimple &am_nnet,
> > +                        const MatrixBase<BaseFloat> &feats,
> > +                        const VectorBase<BaseFloat> *ivector = NULL,
> > +                        const MatrixBase<BaseFloat> *online_ivectors = NULL,
> > +                        int32 online_ivector_period = NULL);
>
> Setting a non-pointer default to NULL. Is it a mistake?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/83/files#r37599543>.
>